validator_client/src/validator_store.rs

```diff
@@ -487,6 +493,28 @@ impl<T: SlotClock + 'static, E: EthSpec> ValidatorStore<T, E> {
         )
     }
 
+    /// Returns a `u64` for the given public key that denotes the builder boost factor. The priority order for fetching this value is:
+    ///
+    /// 1. validator_definitions.yml
+    /// 2. process level flag
+    pub fn get_builder_boost_factor(&self, validator_pubkey: &PublicKeyBytes) -> Option<u64> {
+        self.validators
+            .read()
+            .builder_boost_factor(validator_pubkey)
+    }
```

Comment:
I think this function is missing the fallback to the process-level configuration?

It just calls `InitializedValidators::builder_boost_factor` which doesn't use the process-level config either: https://github.com/sigp/lighthouse/blob/c852a51302a256d9c305fccfb3e4eb94c6c4b5ad/validator_client/src/initialized_validators.rs#L824-L828

Reply:
Looking at the current logic, it looks like the process wide `prefer_builder_proposals` value would override per validator `builder_boost_factor`. It probably makes more sense that per-validator config overrides CLI flags. I've drafted some changes here:
https://github.com/jimmygchen/lighthouse/commit/03152d20f3951fd4cca1c2e35a73c06be1cb0bb4

Going tidy it up now, and add some tests, then push it to this branch.

Reply:
Updated in https://github.com/sigp/lighthouse/pull/5086/commits/b225328f9dd6a0bcfcfe83229b58d8774f9771d5

Per-validator config is now prioritised over process level configurations. I'm looking at potentially adding an http test for testing this logic, but may require a bit of refactor of the tests.

Reply:
Testing all the permutations is a bit tricky with the current setup - we'll need to run a `ProductionValidatorClient` or make `BlockService` a bit more testable.

We could potentially introduce some mocking libraries, or add some abstractions to improve the validator client testability - @michaelsproul and I chatted about this a while ago, but I couldn't find where our discussion was.

There's a bit of setup required for this so might have to leave it out of this PR.